src/sonic-yang-models/yang-models/sonic-device_neighbor_metadata.yang

```diff
@@ -0,0 +1,79 @@
+module sonic-device_neighbor_metadata {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-device_neighbor_metadata";
+    prefix device_neighbor_metadata;
+
+    import ietf-yang-types {
+        prefix yang;
+    }
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    description "DEVICE_NEIGHBOR_METADATA YANG Module for SONiC OS";
+
+    revision 2022-08-25 {
+        description "First Revision";
+    }
+
+    container sonic-device_neighbor_metadata {
+
+        container DEVICE_NEIGHBOR_METADATA {
+
+            description "DEVICE_NEIGHBOR_METADATA part of config_db.json";
+
+            list DEVICE_NEIGHBOR_METADATA_LIST {
+
+                key "name";
+
+                leaf name {
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf lo_addr {
+                    type inet:ipv4-prefix;
+                }
+
+                leaf lo_addr_v6 {
+                    type inet:ipv6-prefix;
+                }
+
+                leaf mgmt_addr {
+                    type inet:ipv4-prefix;
+                }
+
+                leaf mgmt_addr_v6 {
+                    type inet:ipv6-prefix;
+                }
+
+                leaf hwsku {
+                    type stypes:hwsku;
+                }
+
+                leaf type {
+                    type string {
+                        length 1..255;
```

Comment:
with fixed pattern, no need for length

Reply:
Thanks, will fix them in next diff. For UT failed, currently the code has some mgmt_addr type issue, which may have both CIDR and Non-CIDR format in config db, I am trying to address them.